nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -165,14 +140,13 @@ protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
     TableReference tr = TableReference.parse(tableIdentifier.name());
     Preconditions.checkArgument(!tr.hasTimestamp(), "Invalid table name: # is only allowed for hashes (reference by " +
         "timestamp is not supported)");
-    UpdateableReference newReference = this.reference;
+    NessieIcebergClient newClient = client;
     if (tr.getReference() != null) {
-      newReference = loadReference(tr.getReference(), tr.getHash());
+      newClient = new NessieIcebergClient(client.getApi(), tr.getReference(), tr.getHash(), catalogOptions);
```

Comment:
Looks like this can become a `forTableReference(tr)` (that also accepts `null`) - then it seems you can get rid of the `catalogOptions` field in this clas.

I.e. the below would become:
```java
    return new NessieTableOperations(
        ContentKey.of(org.projectnessie.model.Namespace.of(tableIdentifier.namespace().levels()), tr.getName()),
        client.forTableReference(tr),,
        fileIO,
        catalogOptions);
```

Reply:
I have another cleanup PR in the pipeline and will adjust this part of the code accordingly there